### ISSUE TITLE
False-positive: Statement could be decoupled from foreach

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#720]
```php
foreach ($foo as $bar) {
    $this->baz(\compact('bar'));
}
```
Looks like `compact` is not detected as `$bar` usage.

**Comment by kalessil** [comment]
Fixed!
